### PR TITLE
Glsl emscripten

### PR DESCRIPTION
**Pull request by axe312ger**
* add glm as dependency so users don't have to grab it on their own
* add some finding i had on another machien to the readme to help others
* fix makefile with our latest findings

### CODE REVIEW

File: src/projectM-emscripten/README [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
 
-Does not work? Try:
+## Troubleshooting
 
-* `mkdir vendor && cp /usr/local/Cellar/glm/xx/include/glm ./vendor/`
-* `emmake make -j4 -lvendor`
+### General
+
+Want to restart the process after pulling or changing config?
+* `rm src/projectM-emscripten/projectW*`
```
Comment (revmischa) [inline src/projectM-emscripten/README]: should be a clean target

File: src/projectM-emscripten/README [hunk 2]
```diff
@@ -1,27 +1,39 @@
 Running on the web.
 
-# Easy way:
+# TLDR:
+* `./autogen.sh`
+* `emconfigure ./configure --enable-emscripten --enable-gles --enable-sdl`
```
Comment (revmischa) [inline src/projectM-emscripten/README]: i tried making it so all you need is `--enable-emscripten` now and it infers the others

File: src/projectM-emscripten/README [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
+* `emmake make -j4 -lvendor`
```
Comment (revmischa) [inline src/projectM-emscripten/README]: is -lvendor needed? that should be in CPPFLAGS or whatever.. `-I`

File: src/projectM-emscripten/README (hunk 1 above)
Comment (axe312ger) [reply src/projectM-emscripten/README]: Is there any way to ensure this like a cleanup script?

**Comment by revmischa** [comment]
@deltaoscarmike chill?

**Comment by deltaoscarmike** [comment]
As is, it doesn't work: 
- FF displays the page but PM doesn't start PM(output log is empty) and finally it freeze
- Chrome displays the page, but PM doesn't start (output log is empty)

You've got it working ?

**Comment by revmischa** [comment]
No, I never got it to fully run. But it's not a regression either; emscripten doesn't work in glsl now. It just gets a little closer to maybe working now
@axe312ger do you know what the current problem is? is it still GLSL versions?

**Comment by deltaoscarmike** [comment]
Oops, i've made a mistake while trying to push a commit on this branch, please delete branch PR92.
Oth, i've got it working (sort of), try this [patch](https://gist.github.com/deltaoscarmike/228535d73718f8abdf851681c68c4cd1)

**Comment by axe312ger** [comment]
Thank you, I gonna have a look!

**Comment by axe312ger** [comment]
Thats my Chrome console output. (It freezes, make sure to open the console first, then navigate to the server running emscripten

screen shot 2018-08-28 at 13 39 26


I guess the shaders are still broken / using the wrong types. I tried to fix this 2 weeks ago but no, no luck. (Well fixing as of randomly changing values that seem to be related and 🤞 )

```
Failed to compile shader 'Fragment: blur1'. Error: ERROR: 0:39: 'assign' : cannot convert from 'const int' to 'mediump float'
Failed to compile shader 'Fragment: blur2'. Error: ERROR: 0:30: '-' : wrong operand types - no operation '-' exists that takes a left-hand operand of type 'const int' and a right operand of type 'mediump float' (or there is no acceptable conversion)
```

I wonder how to add presets to the wasm as well:
```
[PresetLoader] ENOENT error. The path "/presets" probably does not exist. "man open" for more info.
```

Thanks for your help 👋

**Comment by deltaoscarmike** [comment]
WebGL is rather strict concerning implicit casts, this [patch](https://gist.github.com/deltaoscarmike/c840cfbe41219a0513f8f9a2af8c3fcb) will fix blur shaders, re-enable WASM and print root content of the virtual filesystem.
For the preset directory issue, i think this [link](http://kripken.github.io/emscripten-site/docs/api_reference/Filesystem-API.html) could be useful on how to embed presets.
My priority is to get a rendering closer to Milkdrop/winamp and that's lot of work :wrench:
Good luck

**Comment by axe312ger** [comment]
It works now!!! Thanks so much @deltaoscarmike 

screen shot 2018-08-29 at 11 25 12

**Comment by axe312ger** [comment]
I fixed the merge conflicts :)

**Comment by axe312ger** [comment]
Still need to read into the file system api for emscripten to get the presets running

**Pull request merged**